azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java

```diff
@@ -87,12 +87,12 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
                     request.getSession().setAttribute(CURRENT_USER_PRINCIPAL_JWT_TOKEN, idToken);
                 }
 
-                final Authentication authentication = new PreAuthenticatedAuthenticationToken(
+                final PreAuthenticatedAuthenticationToken preAuth = new PreAuthenticatedAuthenticationToken(
                         principal, null, client.convertGroupsToGrantedAuthorities(principal.getUserGroups()));
-
-                authentication.setAuthenticated(true);
-                log.info("Request token verification success. {}", authentication);
-                SecurityContextHolder.getContext().setAuthentication(authentication);
+                preAuth.setDetails(graphApiToken);
+                preAuth.setAuthenticated(true);
+                log.info("Request token verification success. {}", preAuth);
```

Comment:
Why do you need to put graphApi Token in PreAuthenticatedAuthenticationToken Object? From my perspective, you shouldn't put graphToken here. If you need this token to invoke graph APIs, you can invoke com.microsoft.azure.spring.autoconfigure.aad.AzureADGraphClient#acquireTokenForGraphApi this method to get the new token, rather than use the old graphToken cached in the Authentication Object.

Reply:
so the reason I had put the token in the Authentication Object was to be able to persist the token in the spring security principal object: 
String token = (String)SecurityContextHolder.getContext().getAuthentication().getDetails();

Reply:
Yeah, I got your reason. But my point is that we shouldn't cache the graphAPIs token here, if you want to use the token, you have many ways to get it, such as getting it from the session, getting it by invoking AzureADGraphClient#acquireTokenForGraphApi. 

Another question: what do you do by using the graphAPIs token? The granted authorities Infos have been put in the context already.